### DISCUSSION TITLE
get random results from service or entities (Medusa V2)

### DISCUSSION THREAD

**Discussion by docloulou** [medusajs/medusa#10388] [Feature Requests]
Hello, 
Is it possible to get random results from query graph or service module without fetching ALL results and sort ... ?
For example, get 10 random products...
Thanks

**Comment by srindom** [comment]
What are you looking to achieve?

**Reply by docloulou** [reply]
On the front: Let’s say I want to display 10 random products (with or without filters) before placing the order. The section would be called: "Have You Thought About This?"

I want to optimize the query by selecting only the 10 random products directly "at the SQL level", without fetching and sorting all products on the application side.

**Reply by docloulou** [reply]
i want something like that:
```
const query = req.scope.resolve(ContainerRegistrationKeys.QUERY);

const { data: products } = await query.graph({
  entity: "product",
  fields: ["id", "title"],
  pagination: {
    limit: 10,
    order: {
      $random: true
    },
  },
});
```

**Comment by damianr13** [comment]
This feature would be useful for me as well
